nifi-commons/nifi-site-to-site-client/src/main/java/org/apache/nifi/remote/client/AbstractPeerPersistence.java

```diff
@@ -44,11 +44,19 @@ protected PeerStatusCache restorePeerStatuses(final BufferedReader reader,
             return null;
         }
 
+        final String remoteInstanceUris;
+        remoteInstanceUris = String.valueOf(reader.readLine());
+        if (remoteInstanceUris == null || !remoteInstanceUris.matches("(.+)://(.+):(.+)/nifi-api(.*)")) {
```

Comment:
I don't think there's any reason to run a regular expression against the URIs - just allow whatever the user entered. Also, that regular expression is not accurate because it assumes that a port must be given, and it may not be. There is also no need to include the `/nifi-api` at the end - the URL that the user enters may be as simple as something like `https://nifi-01`. So I would avoid checking anything like that. Instead, I would suggest that when writing out the line, we write out something like "Remote Instance URI: <uris>". Then when reading it in, we can check if the line starts with "Remote Instance URI: ". And/or I would suggest that the second line should actually be something along the lines of "Encoding-Version: 2". That would make it much simpler in the future to modify this code, by simply reading in the Encoding Version and then we know what to expect.

Reply:
I added a variable REMOTE_INSTANCE_URIS_PREFIX as a prefix and checked if the line starts with it when restoring the peers.
the variable used when writing/restoring and unit test